### ISSUE TITLE
`line-is-absent` lint falsely reports objects with duplicate names in different scopes

### ISSUE THREAD

**Issue by ImilB** [objectionary/lints#1155]
## Problem

The `line-is-absent` lint uses a global key by object name.

In `src/main/resources/org/eolang/lints/refs/line-is-absent.xsl`, the lookup is based on `@name` only, and it does not take lexical scope into account.

## What happens

If an XMIR document contains two objects with the same name in different scopes, and one of them has no `@line`, a reference to the other one may still trigger `line-is-absent`.

That creates a false positive: the lint reports a missing `@line` on an unrelated object that is not the actual resolution target.

## Why this is a bug

The rule is supposed to verify the referenced object, not any object in the document that happens to share the same name.

A global key by `@name` is not enough when the same name is reused in different lexical scopes.

## What should happen

The lint should only report `line-is-absent` when the object actually resolved by the reference lacks `@line`.

Objects with the same name in unrelated scopes should not affect the result.

**Comment by volodya-lombrozo** [comment]
@ImilB do you have any proofs? code examples? Failed test?

**Comment by ImilB** [comment]
Yes. I checked it with a minimal XMIR example and the lint reports a false positive.

Example:

```xml
<object author="tests">
  <o>
    <o name="foo" line="1"/>
    <o name="bar" base="Φ.foo" line="2"/>
  </o>
  <o>
    <o name="foo"/>
  </o>
</object>
the current line-is-absent lint reports:
The @line attribute is absent at foo

**Comment by h1alexbel** [comment]
@volodya-lombrozo I'm in

**Comment by volodya-lombrozo** [comment]
@ImilB And why it's a wrong object?

**Comment by ImilB** [comment]
Because `bar` is located in the same local scope as the first `foo`, and that `foo` already has `line="1"`.

So if `Φ.foo` resolves to the local object, then the reported defect is pointing to a different `foo` with the same name from another scope.


The current rule looks objects up globally by `@name` only, so it does not distinguish between these two `foo` objects.

**Comment by volodya-lombrozo** [comment]
@ImilB We have the following XMIR:

```xml
<object author="tests">
  <o>
    <o name="foo" line="1"/>
    <o name="bar" base="Φ.foo" line="2"/>
  </o>
  <o>
    <o name="foo"/>
  </o>
</object>
```

And the following error message:

```
The @line attribute is absent at foo
```

To my mind it's a perfectly valid message for this XMIR. The `foo` object doesn't have a line, indeed.

**Comment by ImilB** [comment]
You're right. This XMIR does not prove the bug I described

The defect message can be valid here, because there is indeed an `foo` object without `@line`. My example was not strict enough to show that the lint points to the wrong object.
Thank you for checking it carefully) So I don't have a confirmed bug report for this case yet. I'll either find a stricter reproducer or treat this issue as unconfirmed.

**Comment by h1alexbel** [comment]
@volodya-lombrozo @ImilB However for this case it looks like a bug:

```xml
<object author="tests">
    <o>
      <o name="foo" line="1"/>
      <o name="bar" base="foo" line="2"/>
    </o>
    <o>
      <o name="foo"/>
    </o>
</object>
```

here, `@base="foo"` refers to closest foo (the same scope), while `missing-line` reports a global one without line:

```text
The @line attribute is absent at foo
```

WDYT?

**Comment by 0crat** [comment]
@imilb Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+18](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by volodya-lombrozo** [comment]
@h1alexbel What would you like to see in the error message? As far as I understand, you have concerns about imprecise message?

**Comment by h1alexbel** [comment]
@volodya-lombrozo as far I understood the semantics of the example above, `line-is-absent` should not report the `foo` here:

```xml
<object author="tests">
  <o>
    <o name="foo" line="1"/>
    <o name="bar" base="foo" line="2"/>
  </o>
  <o>
    <o name="foo"/>
  </o>
</object>
```

`bar` has a `@base="foo"`, which refers to the first `foo` that has `@line="1"`, not the second one (lint should be silent about it). WDYT?

**Comment by volodya-lombrozo** [comment]
@h1alexbel based on the lint description, lines should be assigned only to objects with `base`. So, maybe you are right that we don't need to fire a warning on `foo`

**Comment by h1alexbel** [comment]
@volodya-lombrozo I suggest to fix this then, WDYT?

**Comment by volodya-lombrozo** [comment]
@h1alexbel sure! Please, go ahead.

**Comment by VasilevNStas** [comment]
This is already addressed. The `line-is-absent` lint was reworked in the #1198 effort (commits 2026-08-04/08-05), which @h1alexbel then closed.

The exact scenario from this report — two objects with the same name in different lexical scopes, where a reference resolves to the one that *has* `@line` — is covered by the pack:

`packs/single/line-is-absent/allows-all-lines-present-in-diff-scopes.yaml`

```xml
<o line="1">
  <o line="2"><o name="foo" line="2"/></o>
  <o line="3"><o name="bar" base="ξ.foo" line="3"/></o>
</o>
```

with `defects: 0`, i.e. the reference to the first `foo` (which has `@line`) does not trigger the lint. Suggest closing as resolved.
